go-controller/pkg/node/gateway_localnet.go

```diff
@@ -156,27 +156,43 @@ func initLocalnetGateway(nodeName string, subnet *net.IPNet, wf *factory.WatchFa
 	if err != nil {
 		return err
 	}
-
-	var gatewayIP, gatewayNextHop net.IP
-	var gatewaySubnetMask net.IPMask
-	if utilnet.IsIPv6CIDR(subnet) {
-		gatewayIP = net.ParseIP(v6localnetGatewayIP)
-		gatewayNextHop = net.ParseIP(v6localnetGatewayNextHop)
-		gatewaySubnetMask = net.CIDRMask(v6localnetGatewaySubnetPrefix, 128)
-	} else {
-		gatewayIP = net.ParseIP(v4localnetGatewayIP)
-		gatewayNextHop = net.ParseIP(v4localnetGatewayNextHop)
-		gatewaySubnetMask = net.CIDRMask(v4localnetGatewaySubnetPrefix, 32)
+	// Flush any addresses on localnetGatewayNextHopPort first
+	if err := util.LinkAddrFlush(link); err != nil {
+		return err
 	}
-	gatewayIPCIDR := &net.IPNet{IP: gatewayIP, Mask: gatewaySubnetMask}
-	gatewayNextHopCIDR := &net.IPNet{IP: gatewayNextHop, Mask: gatewaySubnetMask}
 
-	// Flush any addresses on localnetBridgeNextHopPort and add the new IP address.
-	if err = util.LinkAddrFlush(link); err == nil {
-		err = util.LinkAddrAdd(link, gatewayNextHopCIDR)
-	}
-	if err != nil {
-		return err
+	var gatewayIPs []*net.IPNet
+	var nextHops []net.IP
+	for _, hostSubnet := range hostSubnets {
+		var gatewayIP, gatewayNextHop net.IP
+		var gatewaySubnetMask net.IPMask
+		if utilnet.IsIPv6CIDR(hostSubnet) {
```

Comment:
So it looks like the `v*localnetGatewayIP`, `v*localnetGatewayNextHop`, and `v*localnetGatewaySubnetPrefix` are constants and the corresponding net.IP and net.IPMask variables can be computed once outside the for loop and inside the loop you can append the v4 flavor vs. v6?

Reply:
There's at most one v4 subnet and at most one v6 subnet though, so in the dual-stack case we do the same amount of work whether we parse them inside or outside the loop, and in the single-stack case we do less work if we parse them inside the loop since we'll only end up parsing one of them